internal/sysfs/sock.go

```diff
@@ -0,0 +1,66 @@
+package sysfs
+
+import (
+	"net"
+	"syscall"
+
+	"github.com/tetratelabs/wazero/internal/fsapi"
+	"github.com/tetratelabs/wazero/internal/platform"
+)
+
+// SockAccept exposes syscall.Accept on POSIX systems.
+func SockAccept(f fsapi.File) (net.Conn, syscall.Errno) {
+	conn, err := f.(*listenerFile).ln.Accept()
+	if err != nil {
+		return nil, platform.UnwrapOSError(err)
+	}
+	return conn, 0
+}
+
+// SockSetNonblock exposes syscall.SetNonblock on POSIX systems.
+func SockSetNonblock(conn net.Conn) syscall.Errno {
+	tcpConn, ok := conn.(*net.TCPConn)
+	if !ok {
+		return syscall.EINVAL // FIXME: better errno?
```

Comment:
I think this should be a panic because our preconditions should have guaranteed this is a TCPConn. I'll look closer to see why this call site is unsure of the type.

Reply:
new code doesn't need to cast anymore